go-controller/pkg/factory/factory.go

```diff
@@ -215,10 +256,15 @@ func (wf *WatchFactory) addHandler(objType reflect.Type, namespace string, lsel
 	inf.Lock()
 	defer inf.Unlock()
 
-	inf.handlers[handlerID] = cache.FilteringResourceEventHandler{
-		FilterFunc: filterFunc,
-		Handler:    funcs,
+	handler := &Handler{
+		cache.FilteringResourceEventHandler{
+			FilterFunc: filterFunc,
+			Handler:    funcs,
+		},
+		handlerID,
+		0,
```

Comment:
s/0/HANDLER_ALIVE just so that is more readable.

Reply:
agreed.